addons/binding/org.openhab.binding.freemobilesms/src/main/java/org/openhab/binding/freemobilesms/internal/FreeMobileSmsHandler.java

```diff
@@ -0,0 +1,123 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.freemobilesms.internal.FreeMobileSmsConfiguration;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import fr.free.smsapi.Account;
+import fr.free.smsapi.Sender;
+import fr.free.smsapi.RawAccount;
+
+/**
+ * The {@link FreeMobileSmsHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Guilhem Bonnefille <guilhem.bonnefille@gmail.com> - Initial contribution
+ */
+@NonNullByDefault
+public class FreeMobileSmsHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FreeMobileSmsHandler.class);
+
+    private FreeMobileSmsConfiguration config;
+
+    private final Sender sender = new Sender();
+
+    private Account account;
+
+    public FreeMobileSmsHandler(Thing thing) {
+        super(thing);
+    }
+
+    protected void sendMessage(String message) {
+      logger.debug("Message to send: {}", message);
+      try {
+        sender.send(account, message);
+      } catch (IOException e) {
+        logger.error("Failed to send message: {}", e.getMessage());
+      }
+    }
+
+    private void handleMessage(Command command) {
+        logger.debug("Message command");
+        if (command instanceof StringType) {
+            sendMessage(command.toString());
+        } else {
+            logger.warn("Unsupported Command type");
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (CHANNEL_MESSAGE.equals(channelUID.getId())) {
+            if (command instanceof RefreshType) {
+                logger.debug("Refresh command");
+                // TODO: handle data refresh
+                return;
+            }
+
+            handleMessage(command);
+
+            // Note: if communication with thing fails for some reason,
+            // indicate that by setting the status with detail information:
+            // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+            // "Could not control device at IP address x.x.x.x");
+        } else {
+          logger.warn("Received command {} for unknown channel: {}", command, channelUID);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // logger.debug("Start initializing!");
+        config = getConfigAs(FreeMobileSmsConfiguration.class);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
```

Comment:
You can do all the config check stuff in the main thread and you should. Because if anything is not correct, just throw an exception and the framework will tell the user  or put the thing offline.

Regarding `scheduler`: Keep the return value. The reference allows you to "cancel" the future when "dispose" is called.

Reply:
refactored